### PR TITLE
KFSPTS-11657: PMW changes to track approve-disapprove action on PVEN.…

### PR DESCRIPTION
**Pull request by nkimble**
… Code is the fix for the reversion performed in KFSPTS-11644.

### CODE REVIEW

**Comment by nkimble** [comment]
The changes in this file are exactly the same as what was in the original commit performed in JIRA KFSPTS-9545 with the following difference: 
~Line 242 "super.doRouteStatusChange(header);" was removed.

**Comment by cah292** [comment]
Did you mean the this.saveBusinessObject() call was removed? That's what I'm seeing in the older version of the feature.

**Comment by nkimble** [comment]
Yes.You are correct. 
"this" instead of "super" as stated in my previous comment. 
Sorry for the confusion.

File: src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java [hunk 1]
```diff
@@ -225,4 +229,81 @@ private void setupMultiselectField(MaintenanceDocument document, MaintenanceDocu
...
+            processKfsVendorEditActionApproval(kfsDocumentNumber, vendorDetail);
+        }
+    }
+    
+    private boolean isExistingPaymentWorksVendor(String kfsDocumentNumber) {
+        return getPaymentWorksBatchUtilityService().foundExistingPaymentWorksVendorByKfsDocumentNumber(kfsDocumentNumber);
+    }
+    
+    private boolean recordPvenRouteStatusChangeForPaymentWorksIsEnabled() {
+        return (getPaymentWorksBatchUtilityService().retrievePaymentWorksParameterValue(PaymentWorksParameterConstants.PMW_INTEGRATION_IS_ACTIVE_IND).equalsIgnoreCase(KFSConstants.ParameterValues.YES) ? true : false);
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java]: You're already getting a boolean result from this check, so adding a ternary operation of "? true : false" is redundant.

Comment (nkimble) [reply src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java]: You are correct. Removing the ternary operator.

**Comment by nkimble** [comment]
Suggested changes has been made. This is ready to be reviewed again.

File: src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java [hunk 2]
```diff
@@ -46,6 +48,8 @@
     private static final String PROC_METHODS_MULTISELECT_FIELD_NAME = "extension.procurementMethodsArray";
     private static final String MULTISELECT_FIELD_PATH_PREFIX = "dataObject.";
     
+    protected PaymentWorksBatchUtilityService paymentWorksBatchUtilityService;
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java]: Should this service be declared as transient?

**Comment by nkimble** [comment]
Suggested change was made for adding transient. This is ready t be reviewed again.

**Pull request merged**